File: convershark-mobile/lib/models/friend.model.dart
class FriendModel {
  final String id;
  final String uid;
  final String status;
  final String wallpaper;
  final String avatar;
  final String bio;
  final String createdAt;

  const FriendModel({
    required this.id,
    required this.uid,
    required this.status,
    required this.wallpaper,
    required this.avatar,
    required this.bio,
    required this.createdAt,
  });

  factory FriendModel.fromJson(Map<String, dynamic> json) {
    return FriendModel(
      id: json['_id'],
      uid: json['_uid'],
      status: json['status'],
      wallpaper: json['wallpaper'],
      avatar: json['avatar'],
      bio: json['bio'],
      createdAt: json['createdAt'],
    );
  }
}

File: convershark-web/src/models/friend.model.ts
import { StatusType } from 'models/user.model';

export interface IFriend {
  _id: string;
  _uid: string;
  name: string;
  email: string;
  status: StatusType;
  wallpaper: string;
  avatar: string;
  bio: string;
  createdAt: string;
}
